#!/usr/bin/env python
PACKAGE = "hector_nav2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("xy_kp", double_t, 0, "xy_kp", 1.0, 0, 3.0)
gen.add("xy_ki", double_t, 0, "xy_ki", 0,   0, 3.0)
gen.add("xy_kd", double_t, 0, "xy_kd", 0,   0, 3.0)
gen.add("xy_limit", double_t, 0, "the limit of xy output", 3, 0, 5)
gen.add("z_kp",  double_t, 0, "z_kp",  1.2, 0, 3.0)
gen.add("z_ki",  double_t, 0, "z_ki",  0,   0, 3.0)
gen.add("z_kd",  double_t, 0, "z_kd",  0,   0, 3.0)
gen.add("z_limit",  double_t, 0, "the limit of z output",  3, 0, 5)
gen.add("yaw_kp",double_t, 0, "yaw_kp",1.8, 0, 3.0)
gen.add("yaw_ki",double_t, 0, "yaw_ki",0,   0, 3.0)
gen.add("yaw_kd",double_t, 0, "yaw_kd",0,   0, 3.0)
gen.add("yaw_limit",double_t, 0, "the limit of yaw output",2, 0, 5)

gen.add("cof_flighttime", double_t, 0, "cof flighttime", 1.3, 0, 5)
gen.add("cof_acce",       double_t, 0, "cof acce",       0.02,0, 5)
gen.add("plan_vel",       double_t, 0, "plan of vel",    0.5, 0, 5)

exit(gen.generate(PACKAGE, "hector_nav2", "param_reconfig"))
